### DISCUSSION TITLE
Permission denied - can't acces folder on HDD

### DISCUSSION THREAD

**Discussion by 3rpi** [abraunegg/onedrive#2540] [Q&A] [answered]
Hi!
I installed the OneDrive client on Ubuntu Server.
I configured everything and it works fine.
I have mounted an 1TB HDD for the backup in the /mnt/HDD1 folder
In the config file I changed the folder:
sync_dir: "/mnt/HDD1/OneDrive" 
I try to sync, but I get "Permission denied" error massage.
First it synced to my home folder, but there is not enough space and there I had no permission issues
I tried to google the problem, but didn't find a solution.

How can I give the app permissions to the folder?
I know I have to use the chown and chmod commands, but I don't know how exactly
(I'm not a poweruser, learning now the linux system. English is not my first language)

**Comment by abraunegg** [comment, marked as answer]
@3rpi 
> How can I give the app permissions to the folder?

What you need to do is step through the permissions of where you are mounting your 1TB disk to.

Usually, by default, `/mnt` is owned by 'root' : 
```
drwxr-xr-x.   2 root root    6 Jun 22  2021 mnt
```

When you 'mount' your 1TB disk as `/mnt/HDD1` it will inherit the permissions of `/mnt` ... so in this case it will inherit `rwxr-xr-x` which translates to:
* owner (root) = rwx = read, write and execute
* group (root) = r-x = read and execute
* world (anyone else) = r-x = read and execute

So only 'root' has 'write' permissions to your mounted 1TB drive .. this is why you get 'Permission denied'.

What you need to do is:
* Change the permissions of `/mnt/HDD1/` so that your user group can have rwx permissions once the 1TB disk is mounted.

When you read the permissions for `/mnt/HDD1/` it should be similar to:
```
drwxrwxr-x.   2 root users    6 Jun 22  2021 HDD1
```

You then need to ensure that your 'user' (who you log in as) is part of the 'users' group.

If you are still struggling with this issue, this is *not* a problem with this client, but the permissions being used by your system (Ubuntu). You will need to seek support from Ubuntu Forums or similar.

**Reply by 3rpi** [reply]
Thank you for your help.
I could write the folder with my user, and only onedrive had problems with permissions. 
I reset the permissions and changed them again, after that it works fine. I don't now right now what the problem was, but it works now. 
I think it was a PEBCAK problem.

**Comment by gitkevin** [comment]
I feel like I'm so close to making use of OneDrive in the driveone/onedrive:edge docker image. 

There's exactly one left ERROR, and it's the permission denied issue shown below.
```
[DEBUG] applicationVersion:       2.4.25
[DEBUG] latestVersion:            2.4.25
[DEBUG] Setting file permissions for: /home/onedrive/.config/onedrive/refresh_token
Opening the item database ...
[DEBUG] Using database file: /home/onedrive/.config/onedrive/items.sqlite3
[DEBUG] Configuring default new folder permissions as: 777
[DEBUG] Configuring default new file permissions as: 666
All operations will be performed in: /home/gitkevin/OneDrive/testing123
[DEBUG] syncDir: Configured syncDir is missing. Creating: /home/gitkevin/OneDrive/testing123
ERROR: Unable to create local OneDrive syncDir - /home/gitkevin: Permission denied
[DEBUG] Exit scope called
[DEBUG] Running cleanupDryRunDatabase
```

Changing permissions didn't make any difference.
```
Config option 'sync_dir_permissions'         = 777
Config option 'sync_file_permissions'        = 666
```

I start OneDrive in docker as below
```docker run -it --name onedrive_test -v /home/gitkevin/.config/onedrive:/etc/onedrive -v /home/gitkevin/OneDrive/piChecks/testing123:/OneDrive/piChecks/testing123 -e ONEDRIVE_UID=59031 -e ONEDRIVE_GID=59031 driveone/onedrive:edge --verbose --synchronize --verbose --single-directory /home/gitkevin/OneDrive/piChecks/testing123```


Any ideas as to what I should try next?

Thanks...

**Reply by abraunegg** [reply]
@gitkevin 
You need to check the permissions of `/home/gitkevin`

What 'user' are you running 'docker' as?

Setting 'sync_dir_permissions' & 'sync_file_permissions' is only for the files that the 'user' running 'docker' as will then use to create new files in a directory that it can create, but if that 'user' cannot create the folders under `/home/gitkevin/` - you need to *fix* your permissions.

You are passing in also '59031' as the UID and GID of what Docker should use as the internal Docker user .. is this the *same* UID and GID as 'gitkevin' ?

**Reply by gitkevin** [reply]
1. You need to check the permissions of /home/gitkevin

    `drwxr-xr-x. 62 gitkevin root 4096 Nov 15 22:27 gitkevin`

    root is a little odd.
    gitkevin is the owner and group for ~gitkevin/OneDrive/ and below.
    The umask of gitkevin is such that directories are created rwxrwxr-x and files are created as rw-rw-r--

2. What 'user' are you running 'docker' as?

    `gitkevin`

3.  Setting 'sync_dir_permissions' & 'sync_file_permissions' is only for the files that the 'user' running 'docker' as will then use to create new files in a directory that it can create, but if that 'user' cannot create the folders under /home/gitkevin/ - you need to fix your permissions.

    `gitkevin's UID and GID are both 59031. gitkevin can create, modify, remove etc, files within the ~gitkevin space on the host (not the container).`

4. You are passing in also '59031' as the UID and GID of what Docker should use as the internal Docker user .. is this the same UID and GID as 'gitkevin' ?

    `Yes, this the same UID and GID as 'gitkevin'.`

If there's any other info you might need, let me know. I saved stdout/stderr from the earlier attempt.

Thank you!

**Reply by abraunegg** [reply]
As per the original advice - you need to fix your permissions on `/home/gitkevin` ... it should not have a group ownership of 'root'

**Comment by gitkevin** [comment]
Done.

`drwxrwxr-x. 62 gitkevin gitkevin 4096 Nov 15 22:27 gitkevin`

The results were identical.

Now I'm noticing a period at the end of the permissions list. That period indicates [SELinux](https://github.com/SELinuxProject) extended attributes. That's something I don't have access to alter. Beyond that, the programs which manipulate that don't exist on my host. I'll report back here when I figure out how to take care of this.

Thanks.

**Reply by abraunegg** [reply]
@gitkevin 
> I start OneDrive in docker as below
> ...
> Now I'm noticing a period at the end of the permissions list. That period indicates [SELinux](https://github.com/SELinuxProject) extended attributes.
> ...

OK .. given your last issue (using the wrong installation instructions) .. I have to ask - have you followed the correct Docker installation process? Please re-read: https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md

![image](https://github.com/abraunegg/onedrive/assets/4956234/a05b4ad6-e6a0-4282-bc41-bf8be2279602)

**Reply by gitkevin** [reply]
I was unaware that the host has SELinux on it anywhere. 

Yes, I did use the Docker.md instructions you referenced, and apparently didn't do the steps completely for SELinux.
